blog/2021/02/julia-1.6-highlights.md

```diff
@@ -0,0 +1,389 @@
+speed improvements seen in subsequent calls. This change highlights that
+behavior, serving as both a reminder and a tool for rooting out unwanted
+compilation effort i.e. over-specialized code.
+
+[^1]: Note that in some cases the system will look inside the `@time` expression and compile some of the called code before execution of the top-level expression begins. When that happens, some compilation time will not be counted. To include this time you can run `@time @eval ...`
+
+## Eliminating needless recompilation
+
+_Tim Holy_
+
+One of Julia’s most powerful features is its extensibility: you can add new
+methods to previously-defined functions, and use previously-defined methods on
+new types. Sometimes, these new entities force Julia to recompile code to
+account for changes in dispatch. This happens in two steps: first, “outdated”
+code gets _invalidated_, marking it as unsuitable for use; second, as needed the
+code is again compiled from scratch taking account of the new methods and types.
+
+Earlier versions of Julia were somewhat conservative, and invalidated old code
+in some circumstances where there was no actual change in dispatch. Moreover,
+there were many places where Julia and its standard libraries were written in a
+way that defeated Julia’s type-inference. Because the compiler sometimes had to
+invalidate code just because a new method _might_ apply, any uncertainty about
+types magnifies the risk and frequency of invalidation. In older versions of
+Julia, the combination of these effects made invalidation widespread: just
+loading certain packages led to invalidation of up to 10% of Julia’s precompiled
+code. The delay for recompilation could sometimes make interactive sessions feel
+sluggish. When invalidation occurred in Julia’s package-loading code, it also
+delayed loading of the next package, contributing to long waits for `using
+SomePkg` when `SomePkg` depends on other packages.
+
+In 1.6, the scheme for invalidating old code has been made more accurate and
+selective. Moreover, Julia and its standard libraries received a thorough
+makeover to help type inference arrive at a concrete answer more often. The
+result is a leaner, faster Julia that is far more impervious to method
+invalidation, and feels considerably more responsive and nimble in interactive
+sessions. Related blog post: [Analyzing sources of compiler latency in Julia:
+method invalidations](https://julialang.org/blog/2020/08/invalidations/).
+
+## Compiler latency reduction
+
+_Jameson Nash and Jeff Bezanson_
+
+In addition to making our library code more compiler-friendly, we continue
+to try to speed up the compiler itself. This remains one of our main
+technical challenges. In this release there aren't any major
+breakthroughs, but we do have some modest improvements due to work on the
+method table data structure.
+
+Method specificity is a partial order, and prior to 1.6 we stored methods in
```

Comment:
```suggestion
Method specificity is stored in partial order, prior to 1.6, we stored methods in
```

Reply:
If we keep the same wording here, perhaps adding a link to: https://en.wikipedia.org/wiki/Partially_ordered_set would help anyone scratching their head as I was